plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
@timothysc @liggitt How about this: change the default scheduler name to "kube-scheduler" instead of "default-scheduler", then use the s.schedulerName as the lock, it will backwards compatible with old schedulers (if there is a new+ old scenario), also can support multi scheduler.